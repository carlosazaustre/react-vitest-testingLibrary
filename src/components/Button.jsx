import styled, { css } from "styled-components";

const Button = styled.button`
    background-color: #f0f0f0;
    border: none;
    border-radius: 4px;
    color: #000;
    cursor: pointer;
    font-size: 16px;
    padding: 10px 20px;
    transition: all 0.3s ease;
    &:hover {
        background-color: #e0e0e0;
    }

    ${props => props.primary && css`
        background-color: #000;
        color: #fff;
    `}
`;

export default Button;