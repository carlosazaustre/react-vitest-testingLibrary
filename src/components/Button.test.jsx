import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('<Button />', () => {
    it('should match snapshot without props', () => {
        render(<Button />);
        expect(screen.getByRole('button')).toMatchSnapshot();
    });

    it('should match snapshot with primary prop', () => {
        render(<Button primary />);
        expect(screen.getByRole('button')).toMatchSnapshot();
    });
});