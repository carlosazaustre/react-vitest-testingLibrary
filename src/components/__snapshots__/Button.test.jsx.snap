// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match snapshot with primary prop 1`] = `
.c0 {
  background-color: #f0f0f0;
  border: none;
  border-radius: 4px;
  color: #000;
  cursor: pointer;
  font-size: 16px;
  padding: 10px 20px;
  transition: all 0.3s ease;
  background-color: #000;
  color: #fff;
}

.c0:hover {
  background-color: #e0e0e0;
}

<button
  class="c0"
/>
`;

exports[`<Button /> > should match snapshot without props 1`] = `
.c0 {
  background-color: #f0f0f0;
  border: none;
  border-radius: 4px;
  color: #000;
  cursor: pointer;
  font-size: 16px;
  padding: 10px 20px;
  transition: all 0.3s ease;
}

.c0:hover {
  background-color: #e0e0e0;
}

<button
  class="c0"
/>
`;
